int32 ts
float32 x
float32 y
float32 height